hera/animations/optionalWinAnimation.tsx

```diff
@@ -0,0 +1,55 @@
+import { OptionalWinActionResponse } from '@deities/apollo/GameOver.tsx';
+import { fbt } from 'fbt';
+import { resetBehavior } from '../behavior/Behavior.tsx';
+import NullBehavior from '../behavior/NullBehavior.tsx';
+import AnimationKey from '../lib/AnimationKey.tsx';
+import getWinCriteriaName from '../lib/getWinCriteriaName.tsx';
+import { Actions, State } from '../Types.tsx';
+
+export default async function optionalWinAnimation(
+  actions: Actions,
+  state: State,
+  actionResponse: OptionalWinActionResponse,
+): Promise<State> {
+  const { requestFrame, update } = actions;
+  const { condition } = actionResponse;
+  const player = state.map.getCurrentPlayer().id;
+  return new Promise((resolve) =>
+    update((state) => ({
+      animations: state.animations.set(new AnimationKey(), {
+        color: player,
+        length: 'medium',
+        onComplete: (state) => ({
+          ...state,
+          animations: state.animations.set(new AnimationKey(), {
+            color: player,
+            length: 'medium',
+            onComplete: (state) => {
+              requestFrame(() => resolve(state));
+              return state;
+            },
+            player,
+            sound: 'UI/Start',
+            style: 'flashy',
+            text: String(getWinCriteriaName(condition.type)),
+            type: 'banner',
+          }),
+          ...resetBehavior(),
+          behavior: new NullBehavior(),
+        }),
+        player,
+        sound: 'UI/Start',
+        text: String(
+          fbt(
+            fbt.param('hidden', condition.hidden ? '(Secret) ' : '') +
+              'Optional Win Condition Fulfilled!',
+            'Optional win banner',
```

Comment:
Actually, let's stick with the "Secret discovered" text for this. Since they are identical except for the text, how about removing this file, and making `secretDiscoveredAnimation` take either an `OptionalWinActionResponse` (after rename `OptionalConditionActionResponse` or a `SecretDiscoveredActionResponse` and handle the text based on the `actionResponse.type`?

For OptionalWin I would expect the text to be "Optional Secret Discovered" if it was hidden, and maybe "Optional Condition Fulfilled" if not secret.

Reply:
Completed in 512e48fd27cb1adda98c1cc27324401947f76420